src/Workspaces/Core/Portable/SemanticModelWorkspaceService/SemanticModelWorkspaceServiceFactory.cs

```diff
@@ -50,13 +50,13 @@ public async Task<SemanticModel> GetSemanticModelForNodeAsync(Document document,
                 if (syntaxFactsService == null || semanticFactsService == null)
                 {
                     // it only works if we can track member
-                    return await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                    return (await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false))!;
```

Comment:
The callers should know this doesn't return null -- this only is usable if you're in a language that supports SemanticModel.

Reply:
i'm not quite getting why this is returning something nullable.  Shouldn't this return non-null, and throw if you call on a language that doesn't support this?

Reply:
The helper this implements will return non-null because this method only supports C#/VB (it's doing all sorts of magic). But the core document.GetSemanticModel returns null because of the F#/TypeScript problem.

Reply:
> But the core document.GetSemanticModel returns null because of the F#/TypeScript problem.

Note sure i'm getting the problem.  There's already a SupportsSemanticModel property.  I'm totally ok saying that the method doens't return null, and it will instead throw if SupportsSemanticModel is false :)

Reply:
Sure, that's a breaking change though. :-)

Reply:
you bring me more tears than anyone else in my life :'(

I would actually be shocked to discover people just calling this and dealing with null.  But i agree, such a change could be done *outside* of this PR.

That said, it's also shows me that `await+!` is just unpleasant.